src/server/game/Entities/Player/Player.cpp

```diff
@@ -5053,20 +4846,28 @@ void Player::DeleteOldCharacters(uint32 keepDays)
 
 void Player::SetMovement(PlayerMovementType pType)
 {
-    WorldPacket data;
+    WorldPackets::ServerPacket* packet = nullptr;
     switch (pType)
     {
-        case MOVE_ROOT:       data.Initialize(SMSG_FORCE_MOVE_ROOT,   GetPackGUID().size()+4); break;
-        case MOVE_UNROOT:     data.Initialize(SMSG_FORCE_MOVE_UNROOT, GetPackGUID().size()+4); break;
-        case MOVE_WATER_WALK: data.Initialize(SMSG_MOVE_WATER_WALK,   GetPackGUID().size()+4); break;
-        case MOVE_LAND_WALK:  data.Initialize(SMSG_MOVE_LAND_WALK,    GetPackGUID().size()+4); break;
+        case MOVE_ROOT:       packet = new WorldPackets::Movement::MoveForceRoot; break;
+        case MOVE_UNROOT:     packet = new WorldPackets::Movement::MoveForceUnroot; break;
+        case MOVE_WATER_WALK: packet = new WorldPackets::Movement::MoveForceWaterWalk; break;
+        case MOVE_LAND_WALK:  packet = new WorldPackets::Movement::MoveForceLandWalk; break;
         default:
             TC_LOG_ERROR("entities.player", "Player::SetMovement: Unsupported move type (%d), data not sent to client.", pType);
             return;
     }
-    data << GetPackGUID();
-    data << uint32(0);
-    GetSession()->SendPacket(&data);
+
+    switch (pType)
+    {
+        case MOVE_ROOT: reinterpret_cast<WorldPackets::Movement::MoveForceRoot*>(packet)->Guid = GetPackGUID(); break;
+        case MOVE_UNROOT: reinterpret_cast<WorldPackets::Movement::MoveForceUnroot*>(packet)->Guid = GetPackGUID(); break;
+        case MOVE_WATER_WALK: reinterpret_cast<WorldPackets::Movement::MoveForceWaterWalk*>(packet)->Guid = GetPackGUID(); break;
+        case MOVE_LAND_WALK: reinterpret_cast<WorldPackets::Movement::MoveForceLandWalk*>(packet)->Guid = GetPackGUID(); break;
+    }
+    
+    GetSession()->SendPacket(packet->Write());
+    delete packet;
```

Comment:
@jackpoz, the delete was missing back then, here it is added (packet is not null at this point)